src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingGraphInterface.java

```diff
@@ -0,0 +1,1063 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller.readthreading;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.gatk.nativebindings.smithwaterman.SWOverhangStrategy;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.Kmer;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.graphs.BaseGraph;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.graphs.KmerSearchableGraph;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.graphs.MultiSampleEdge;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.read.AlignmentUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+import org.broadinstitute.hellbender.utils.smithwaterman.SmithWatermanAligner;
+import org.broadinstitute.hellbender.utils.smithwaterman.SmithWatermanAlignment;
+import org.jgrapht.EdgeFactory;
+
+import java.io.File;
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Read threading graph class intended to contain duplicated code between {@link ReadThreadingGraph} and {@link JunctionTreeLinkedDeBruinGraph}.
+ *
+ *
+ */
+public abstract class ReadThreadingGraphInterface extends BaseGraph<MultiDeBruijnVertex, MultiSampleEdge> implements KmerSearchableGraph<MultiDeBruijnVertex,MultiSampleEdge> {
+    private static final long serialVersionUID = 1l;
+    private static final String ANONYMOUS_SAMPLE = "XXX_UNNAMED_XXX";
+    private static final boolean WRITE_GRAPH = false;
+    private static final boolean DEBUG_NON_UNIQUE_CALC = false;
+    private static final int MAX_CIGAR_COMPLEXITY = 3;
+    private static final boolean INCREASE_COUNTS_BACKWARDS = true;
+    /** for debugging info printing */
+    private static int counter = 0;
+    /**
+     * Sequences added for read threading before we've actually built the graph
+     */
+    protected final Map<String, List<SequenceForKmers>> pending = new LinkedHashMap<>();
+    /**
+     * A map from kmers -> their corresponding vertex in the graph
+     */
+    protected final Map<Kmer, MultiDeBruijnVertex> kmerToVertexMap = new LinkedHashMap<>();
+    protected final boolean debugGraphTransformations;
+    protected final byte minBaseQualityToUseInAssembly;
+    protected boolean startThreadingOnlyAtExistingVertex = false;
+    protected List<MultiDeBruijnVertex> referencePath = null;
+
+    private int maxMismatchesInDanglingHead = -1;
+    protected boolean alreadyBuilt;
```

Comment:
Why not initialize this here?

Reply:
Ive pushed most of the `resetToInitialState` logic into these initializations now